packages/core/status/core-status-server-internal/src/log_plugins_status.test.ts

```diff
@@ -0,0 +1,583 @@
+    });
+  });
+});
+
+describe('getPluginStatusChangesMessages', () => {
+  it('returns a list of human readable messages describing the changes', () => {
+    expect(
+      getPluginStatusChangesMessages({
+        status: {
+          available: [
+            {
+              level: ServiceStatusLevels.available,
+              pluginName: 'newlyAvailablePlugin',
+              summary: 'summary',
+              isReportedStatus: true,
+            },
+            {
+              level: ServiceStatusLevels.available,
+              pluginName: 'newlyAvailablePluginDependantPlugin1',
+              summary: 'summary',
+            },
+            {
+              level: ServiceStatusLevels.available,
+              pluginName: 'newlyAvailablePluginDependantPlugin1',
+              summary: 'summary',
+            },
+            {
+              level: ServiceStatusLevels.available,
+              pluginName: 'alreadyAvailablePlugin',
+              summary: 'summary',
+              isReportedStatus: true,
+            },
+            {
+              level: ServiceStatusLevels.available,
+              pluginName: 'alreadyAvailablePluginDependantPlugin1',
+              summary: 'summary',
+            },
+          ],
+
+          critical: [
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'alreadyCriticalPlugin',
+              summary: 'summary',
+              isReportedStatus: true,
+            },
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'alreadyCriticalPluginDependantPlugin1',
+              summary: 'summary',
+            },
+          ],
+
+          degraded: [
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'newlyDegradedPlugin',
+              summary: 'summary',
+              isReportedStatus: true,
+            },
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'newlyDegradedPluginDependantPlugin1',
+              summary: 'summary',
+            },
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'anotherNewlyDegradedPlugin',
+              summary: 'summary',
+              isReportedStatus: true,
+            },
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'anotherNewlyDegradedPluginDependantPlugin1',
+              summary: 'summary',
+            },
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'alreadyDegradedPlugin',
+              summary: 'summary',
+              isReportedStatus: true,
+            },
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'alreadyDegradedPluginDependantPlugin1',
+              summary: 'summary',
+            },
+          ],
+
+          unavailable: [],
+        },
+        total: 2,
+        updated: 2,
```

Comment:
I know this is a test fixture, but are these numbers correct here? Or am I misunderstanding what they mean

Reply:
Good catch! I wrote them in the beginning and then forgot to update them when I added more entries. Will update.